template/src/core/util/hook/useOnUnmount.tsx

```diff
@@ -0,0 +1,15 @@
+import {useLayoutEffect} from "react";
+
+function useOnUnmount(callback?: VoidFunction) {
+  useLayoutEffect(
+    () => () => {
+      if (callback) {
+        callback();
+      }
+    },

```

Comment:
This is much easier to read for me.
```suggestion
  useLayoutEffect(() => {
     return () => {
        if (callback) {
          callback();
        }
    }
  }, []);
```